wlauto/common/linux/workload.py

```diff
@@ -0,0 +1,146 @@
+#    Copyright 2017 ARM Limited
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import os
+import sys
+from math import ceil
+import time
+
+from wlauto.core.extension import Parameter
+from wlauto.core.workload import Workload
+from wlauto.core.resource import NO_ONE
+from wlauto.common.resources import File, Executable
+from wlauto.common.android.resources import ReventFile
+from wlauto.exceptions import WorkloadError, ResourceError, DeviceError
+from wlauto.utils.revent import ReventRecording
+
+class ReventWorkload(Workload):
+    # pylint: disable=attribute-defined-outside-init
+
+    parameters = [
+        Parameter('default_naming', kind=bool, default=False,
+                  description='Use the default naming. Set this option to true if the revent file you wish to use was '
+                              'generated using the wa record command.'),
+        Parameter('idle', kind=bool, default=False,
```

Comment:
You don't need this. Just remove the default for `idle_time`. Then if it is not set (i.e. is `None`), you can assume that `idle == False`.